File: HandOfStraights.py
# https://leetcode.com/problems/hand-of-straights/
class Solution(object):
    def isNStraightHand(self, hand, groupSize):
        countMap = {}
        for card in hand:
            countMap[card] = countMap.get(card, 0) + 1
        hand.sort()
        for i in range(len(hand)):
            if countMap[hand[i]] == 0:
                continue
            for j in range(groupSize):
                currCard = hand[i] + j
                if countMap.get(currCard, 0) == 0:
                    return False
                countMap[currCard] -= 1
        return True

File: HandOfStraights.java
// https://leetcode.com/problems/hand-of-straights/
class Solution {
    public boolean isNStraightHand(int[] hand, int groupSize) {
        Map<Integer, Integer> countMap = new HashMap<>();
        for (int card : hand) {
            countMap.put(card, countMap.getOrDefault(card, 0) + 1);
        }
        Arrays.sort(hand);
        for (int i = 0; i < hand.length; i++) {
            if (countMap.get(hand[i]) == 0) {
                continue;
            }
            for (int j = 0; j < groupSize; j++) {
                int currCard = hand[i] + j;
                if (countMap.getOrDefault(currCard, 0) == 0) {
                    return false;
                }
                countMap.put(currCard, countMap.get(currCard) - 1);
            }
        }
        return true;
    }
}